skore/src/skore/project/project.py

```diff
@@ -103,174 +108,99 @@ def put(self, key: str, value: Any, *, note: Optional[str] = None):
-            return item.series
-        elif isinstance(item, SklearnBaseEstimatorItem):
-            return item.estimator
-        elif isinstance(item, CrossValidationItem):
-            return item.cv_results_serialized
-        elif isinstance(item, MediaItem):
-            return item.media_bytes
-        else:
-            raise ValueError(f"Item {item} is not a known item type.")
+        self.item_repository.put_item(
+            key,
+            object_to_item(
+                value,
+                note=note,
+                display_as=display_as,
+            ),
+        )
 
-    def get_item(self, key: str) -> Item:
-        """Get the item corresponding to ``key`` from the Project.
+    def get(self, key, *, latest=True, metadata=False):
+        """Get the value associated to ``key`` from the Project.
 
         Parameters
         ----------
         key : str
             The key corresponding to the item to get.
+        latest : boolean, optional
+            Get the latest value or all the values associated to ``key``, default True.
+        metadata : boolean, optional
+            Get the metadata in addition of the value, default False.
 
         Returns
         -------
-        item : Item
-            The Item corresponding to ``key``.
+        value : any
+            Value associated to ``key``, when latest=True and metadata=False.
+        value_and_metadata : dict
+            Value associated to ``key`` with its metadata, when latest=True and
+            metadata=True.
+        list_of_values : list[any]
+            Values associated to ``key``, ordered by date, when latest=False.
+        list_of_values_and_metadata : list[dict]
+            Values associated to ``key`` with their metadata, ordered by date, when
+            latest=False and metadata=False.
 
         Raises
         ------
         KeyError
-            If the key does not correspond to any item.
+            If the key is not in the project.
         """
-        return self.item_repository.get_item(key)
+        if not metadata:
 
-    def get_item_versions(self, key: str) -> list[Item]:
-        """
-        Get all the versions of an item associated with ``key`` from the Project.
+            def dto(item):
+                return item_to_object(item)
 
-        The list is ordered from oldest to newest :func:`~skore.Project.put` date.
+        else:
 
-        Parameters
-        ----------
-        key : str
-            The key corresponding to the item to get.
+            def dto(item):
+                return {
+                    "value": item_to_object(item),
+                    "date": item.updated_at,
+                    "note": item.note,
+                }
 
-        Returns
-        -------
-        list[Item]
-            The list of items corresponding to ``key``.
+        if latest:
+            return dto(self.item_repository.get_item(key))
+        return list(map(dto, self.item_repository.get_item_versions(key)))
```

Comment:
nitpick: As previously stated, I think this function does too much. A particularly bad smell is that the return type depends on argument values. We'll see with usage I guess...